autospec/license.py

```diff
@@ -99,7 +99,11 @@ def license_from_copying_hash(copying, srcdir):
         # This is not a text file
         return
 
-    data = data.decode(license_charset)
+    try:
+        data = data.decode(license_charset)
```

Comment:
Hmm, if `license_charset` is set to `ISO-8859-13` or `mac_roman` (see code a few lines above), I wouldn't expect a `UnicodeDecodeError` for those cases...

Reply:
If the code above these lines was working, then this will continue to work.  However, the automatic charset detection failed in this case (it's an ISO file, and charset thinks it's Windows-1251 IIRC), causing `data.decode()` to fail.

Reply:
Yes, I see in this case `charset.detect` detected `Windows-1254`, but with a relatively low confidence score -- approximately `0.46` for `test/data/copying-rr.iso`. `1.0` is the highest score. Print out the value of `detected` to see the details.

We do not inspect the confidence score, but perhaps we should, and filter out files where the confidence is below a certain threshold?

Reply:
After collecting more data, I see that some legitimate license files can have a low confidence score (e.g. `libs/icu/icu-src/LICENSE` from the texlive package is `ISO-8859-1` with a confidence of `0.73`), but other non-license files may have similar score (e.g. `test/data/copying.iso` from libcdio is `ISO-8859-1` with a confidence of `0.7291`). So, I don't trust the chardet confidence ratings enough for autospec to rely on them.

Reply:
Thanks for looking into this.  I suspected that this would be the case, given the heuristics  to change the detected charset to `ISO-8859-13` and `mac_roman` above.  Maybe those could be moved to the `except` block below, and only print the warning/return if those are not contemplated by the `if license_charset ...` block?

Reply:
@lpereira Yes, I like that idea.